File: backend/spec/requests/api/v1/homes_spec.rb
require 'rails_helper'

RSpec.describe "Api::V1::Homes", type: :request do
  describe "GET /top" do
    it "topページを表示する" do
      date_spot = FactoryBot.create(:address).date_spot
      other_spot = FactoryBot.create(:other_address).date_spot
      get "/api/v1/top"
      areas = Area.all.map do |area|
        {
          "attributes"=>{
            "id" => area.id,
            "name" => area.name
          }
        }
      end
      main_prefectures = Prefecture.find([13, 27, 40, 14, 23, 26]).map do |prefecture|
        {
          "attributes"=>{
            "id"=>prefecture.id,
            "name"=>prefecture.name,
            "area_id"=>prefecture.area_id
          }
        }
      end
      main_genres = Genre.find([1, 2, 3, 4, 5, 6]).map do |genre|
        {
          "attributes"=>{
            "id"=>genre.id,
            "name"=>genre.name
          }
        }
      end

      genres = Genre.all.map do |genre|
        {
          "attributes"=>{
            "id"=>genre.id,
            "name"=>genre.name
          }
        }
      end

      expect(JSON.parse(response.body)["areas"]).to eq(areas)
      expect(JSON.parse(response.body)["main_prefectures"]).to eq(main_prefectures)
      expect(JSON.parse(response.body)["main_genres"]).to eq(main_genres)
      expect(JSON.parse(response.body)["genres"]).to eq(genres)
      expect(JSON.parse(response.body)["address_and_date_spots"][0]["date_spot"]["name"]).to eq(date_spot.name)
      expect(JSON.parse(response.body)["address_and_date_spots"][1]["date_spot"]["name"]).to eq(other_spot.name)
    end
  end
end

File: e2e/cypress/integration/homes.spec.ts
import { apiHomeTopAccess } from "../support/backendAccessMock/homes/apiHomeAccess";
import { Genre, Prefecture } from "../support/types/datas/activeHash";

const mainPrefectures: {attributes: Prefecture}[] = [
  { attributes: { id: 13, name: '東京都', areaId: 2 } }, { attributes: {id: 14, name: '神奈川県', areaId: 2 }},
  { attributes: { id: 23, name: '愛知県', areaId: 3 } }, { attributes: { id: 26, name: '京都府', areaId: 4 }},
  { attributes: { id: 27, name: '大阪府', areaId: 4 } }, { attributes: { id: 40, name: '福岡県', areaId: 6 }}
];

const mainGenres: {attributes: Genre}[] = [
  { attributes: { id: 1, name: 'ショッピングモール' } }, { attributes: { id: 2, name: '飲食店' }}, { attributes: { id: 3, name: 'カフェ' } },
  { attributes: { id: 4, name: 'アウトドア' } }, { attributes: { id: 5, name: '遊園地' }}, {attributes: { id: 6, name: '水族館' } },
];

describe('homes', () => {
  it('Topページを表示する', () => {
    apiHomeTopAccess();
    cy.visit('/');
    cy.contains('DateCourses');
    cy.contains('全国の人気ランキング');
    mainPrefectures.map((data) => cy.contains(data.attributes.name));
    mainGenres.map((data) => (cy.contains(data.attributes.name)));
  });
});